### DISCUSSION TITLE
Elastic fills up too quickly

### DISCUSSION THREAD

**Discussion by myrsecurity** [telekom-security/tpotce#1668] [Q&A]
so the partition despite has 400GB assigned to run, has only picked up 100GB

![image](https://github.com/user-attachments/assets/ab0f3a11-bc11-4900-a48f-2e740604d44e)

Not sure where the other space is. Tried gparted live and the remaining 300GB are in the disk, but when ubuntu runs only sees 100GB

Tpot fills this very quickly within days

anything I can do to expand ?

**Comment by dmille6** [comment]
this is how ubuntu defaults its setup for harddrive space. it can easily be expanded.  from the terminal. 

commands/process to do so:

```
sudo su
cfdisk
pvresize /dev/sda3
lvextend -l +100%FREE /dev/ubuntu-vg/ubuntu-lv
resize2fs /dev/mapper/ubuntu--vg-ubuntu--lv
```

Explanation: 
Here’s a breakdown of the commands you've listed:

1. **`sudo su`**:
   - `sudo` allows a permitted user to execute a command as the superuser or another user.
   - `su` stands for "substitute user" and is typically used to switch to the superuser (root). 
   - Combined (`sudo su`), it elevates your permissions to root, giving you full administrative privileges on the system. You can now run any command with root access.

2. **`cfdisk`**:
   - `cfdisk` is a partition editor in Linux that provides a user-friendly interface for managing disk partitions. It's useful for creating, deleting, resizing, and managing partitions on a hard drive.

3. **`pvresize /dev/sda3`**:
   - This command is used to resize a physical volume in an LVM (Logical Volume Manager) setup.
   - `/dev/sda3` is the device (a specific partition of the hard drive) being resized. 
   - `pvresize` adjusts the size of the physical volume to match any changes in the underlying partition, making additional space available for use by LVM.

4. **`lvextend -l +100%FREE /dev/ubuntu-vg/ubuntu-lv`**:
   - This command extends a logical volume to use all available free space in the volume group.
   - `-l +100%FREE` tells the system to use all unallocated space within the volume group.
   - `/dev/ubuntu-vg/ubuntu-lv` specifies the path to the logical volume you're extending, where `ubuntu-vg` is the volume group and `ubuntu-lv` is the logical volume.

5. **`resize2fs /dev/mapper/ubuntu--vg-ubuntu--lv`**:
   - This command resizes the filesystem on the specified logical volume to match its newly expanded size.
   - `resize2fs` is commonly used with ext2, ext3, or ext4 filesystems.
   - `/dev/mapper/ubuntu--vg-ubuntu--lv` is the path to the logical volume that you just extended. The double dashes (`--`) represent a hyphen in the volume and group names.
  
In summary, these commands are used to manage disk partitions, resize physical and logical volumes, and then resize the filesystem in a Linux system, typically in an LVM environment.
